nav2_costmap_2d/include/nav2_costmap_2d/observation.hpp

```diff
@@ -60,9 +60,32 @@ class Observation
   }
 
   /**
-    * @brief  Explicitly define copy assignment operator for Observation as it has a user-declared destructor
-    */
-  Observation & operator=(const Observation &) = default;
+  * @brief  Copy assignment operator
+  * @param obs The observation to copy
+  */
+  Observation & operator=(const Observation & obs){
```

Comment:
This looks fine to me, but I don't think any of the other changes are required.

Reply:
They're indeed not required, but they improve performance a bit.